### PR TITLE
feat(api): enforce inbound User-Agent and propagate shared UA to upstream fetches

### PR DESCRIPTION
**Pull request by DysektAI**
## Summary

- Gateway now rejects API requests with a missing or out-of-range (5-200 char) `User-Agent` header with `400`. This makes the inbound UA visible in usage reporting meaningful and discourages anonymous scrapers.
- OpenAPI spec bumped to `2.3.0`; documents the required `User-Agent` header (component parameter `UserAgentHeader`) and the `invalidUserAgent` bad-request example.
- Server-side fetchers (`tarkov-json`, `overlay`, `twitch/live`, `tarkov-dev/profile`) now send the shared `TARKOVTRACKER_USER_AGENT` constant instead of ad-hoc strings. The hardcoded `PROFILE_FETCH_USER_AGENT` constant in `profile.get.ts` is removed in favor of the shared value.
- Exports `INBOUND_USER_AGENT_MIN_LENGTH` from `workers/api-gateway/src/utils/userAgent.ts` and adds unit tests for `normalizeInboundUserAgent` and the new min/max bounds.

Restores the genuinely new half of the user-agent work that was stranded in an orphaned worktree (PR #555 already landed the storage/migration/recording half). The regressive bits from the old worktree base were dropped.

#### Test plan

- [x] `pnpm run test:api-gateway` — 116 tests pass (includes 2 new UA-gate tests + 6 new `userAgent.test.ts` tests)
- [x] `pnpm run typecheck` — passes
- [x] `pnpm run lint` — zero warnings
- [x] `pnpm run validate:openapi` — schema valid at 2.3.0
- [x] `pnpm vitest run app/server/utils/__tests__/tarkov-json.test.ts app/server/utils/__tests__/overlay.test.ts app/server/api/tarkov-dev/__tests__/profile.test.ts` — 48 tests pass

<a href="https://cubic.dev/pr/tarkovtracker-org/TarkovTracker/pull/572?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

___

## **CodeAnt-AI Description**
**Require a valid User-Agent on API requests and reuse one shared client agent for upstream fetches**

### What Changed
- API requests now fail with `400` when the `User-Agent` header is missing, too short, or too long, with a clear error message.
- The API docs now show `User-Agent` as a required header and include the new bad-request example.
- User-facing server requests to Tarkov.dev, Twitch, overlay data, and Tarkov.dev profiles now send the shared TarkovTracker user agent instead of separate values.
- Usage tracking now records the normalized inbound `User-Agent` for valid requests.

### Impact
`✅ Cleaner API usage reporting`
`✅ Fewer anonymous or malformed API requests`
`✅ Consistent upstream request identity`
<details><summary><strong>💡 Usage Guide</strong></summary>

### Checking Your Pull Request
Every time you make a pull request, our system automatically looks through it. We check for security issues, mistakes in how you're setting up your infrastructure, and common code problems. We do this to make sure your changes are solid and won't cause any trouble later.

### Talking to CodeAnt AI
Got a question or need a hand with something in your pull request? You can easily get in touch with CodeAnt AI right here. Just type the following in a comment on your pull request, and replace "Your question here" with whatever you want to ask:
<pre>
<code>@codeant-ai ask: Your question here</code>
</pre>
This lets you have a chat with CodeAnt AI about your pull request, making it easier to understand and improve your code.

#### Example
<pre>
<code>@codeant-ai ask: Can you suggest a safer alternative to storing this secret?</code>
</pre>

### Preserve Org Learnings with CodeAnt
You can record team preferences so CodeAnt AI applies them in future reviews. Reply directly to the specific CodeAnt AI suggestion (in the same thread) and replace "Your feedback here" with your input:
<pre>
<code>@codeant-ai: Your feedback here</code>
</pre>
This helps CodeAnt AI learn and adapt to your team's coding style and standards.

#### Example
<pre>
<code>@codeant-ai: Do not flag unused imports.</code>
</pre>

### Retrigger review
Ask CodeAnt AI to review the PR again, by typing:
<pre>
<code>@codeant-ai: review</code>
</pre>

### Check Your Repository Health
To analyze the health of your code repository, visit our dashboard at [https://app.codeant.ai](https://app.codeant.ai). This tool helps you identify potential issues and areas for improvement in your codebase, ensuring your repository maintains high standards of code health.

</details>

## Summary by CodeRabbit

- **New Features**
  - Protected API requests now require a valid `User-Agent` header (5–200 characters) and are rejected with a clear `400 Bad Request` when missing/invalid.
  - Outbound requests to external services now consistently include the same application `User-Agent`.
- **Documentation**
  - Updated API documentation and published OpenAPI spec v2.3.0 to document the header requirement, exemptions, and `400` error responses.
  - Updated contributor guidance for required `User-Agent` usage in public progress clients.
- **Tests**
  - Expanded/adjusted gateway and endpoint tests to validate inbound header enforcement and outbound header behavior.

<details open><summary><h3>Confidence Score: 5/5</h3></summary>

Safe to merge — the UA gate logic is correct for all boundary conditions, infrastructure exempt paths are returned before the check, and the OpenAPI spec now properly references the parameter on every protected operation.

The gate correctly handles all boundary cases: null/blank/oversized inputs return null from normalizeInboundUserAgent and are rejected; short strings pass the function but are caught by the length-comparison at the gate; valid strings flow through. The redirect-before-auth ordering is preserved and the new test suite covers the main rejection paths.

No files require special attention. The duplicate TARKOVTRACKER_USER_AGENT constant across workers/ and app/ is a minor maintenance note but does not affect correctness today.
</details>

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| workers/api-gateway/src/utils/userAgent.ts | Now returns null for overlong inputs (fixes prior truncation behaviour) and exports both length bounds; logic is clean and correct |
| workers/api-gateway/src/index.ts | UA validation placed before redirect and auth logic; infrastructure early-returns correctly precede the check; userAgent passed to authenticateAndRateLimit for usage recording |
| workers/api-gateway/src/openapi.ts | Version bumped to 2.3.0; UserAgentHeader parameter now correctly referenced via $ref on each protected operation; 400 response added |
| workers/api-gateway/src/__tests__/userAgent.test.ts | New unit test file covers null/empty/whitespace, trimming, and overlong inputs; missing an at-boundary test for exactly INBOUND_USER_AGENT_MAX_LENGTH chars |
| workers/api-gateway/src/__tests__/gateway.test.ts | buildRequest helper injects default UA; new tests cover missing, too-short, oversized, and pre-redirect UA rejection paths |
| app/server/api/tarkov-dev/profile.get.ts | Local PROFILE_FETCH_USER_AGENT removed; now uses shared TARKOVTRACKER_USER_AGENT from app/server/utils/userAgent.ts |
| app/server/utils/tarkov-json.ts | TarkovJsonRequest.headers type widened to Record<string,string>; User-Agent added to outbound fetch; consistent with other fetchers |
| app/server/utils/overlay.ts | User-Agent header added to overlay fetch using the shared constant; straightforward one-line change |
| app/server/api/twitch/live.get.ts | User-Agent added to Twitch GQL POST using shared constant; test updated to assert the new header |

</details>

<details><summary><h3>Sequence Diagram</h3></summary>

<a href="#gh-light-mode-only">

```mermaid
%%{init: {'theme': 'neutral'}}%%
sequenceDiagram
    participant C as API Client
    participant GW as api-gateway (index.ts)
    participant UA as normalizeInboundUserAgent
    participant Auth as authenticateAndRateLimit
    participant Usage as recordUsage

    C->>GW: HTTP request (any protected path)
    GW->>GW: Resolve apiPath (infra routes exit early)
    GW->>UA: normalizeInboundUserAgent(User-Agent header)
    UA-->>GW: null (missing/blank/oversized) OR trimmed string

    alt "null OR length < 5"
        GW-->>C: 400 User-Agent must be 5-200 characters
    else valid UA
        GW->>GW: Check LEGACY_API_REDIRECT 308 if set
        GW->>Auth: authenticateAndRateLimit(..., inboundUserAgent)
        Auth->>Usage: recordUsage userAgent
        Auth-->>GW: AuthSuccess or 401/403/429
        GW-->>C: 200 / error response
    end
```

</a>
<a href="#gh-dark-mode-only">

```mermaid
%%{init: {'theme': 'base', 'themeVariables': {"darkMode": true, "background": "#0d1117", "primaryColor": "#21262d", "primaryTextColor": "#e6edf3", "primaryBorderColor": "#8b949e", "lineColor": "#8b949e", "textColor": "#e6edf3", "edgeLabelBackground": "#161b22", "actorBkg": "#21262d", "actorBorder": "#8b949e", "actorTextColor": "#e6edf3", "actorLineColor": "#8b949e", "signalColor": "#8b949e", "signalTextColor": "#e6edf3", "noteBkgColor": "#373320", "noteBorderColor": "#d4a72c", "noteTextColor": "#f0e6c0", "labelBoxBkgColor": "#21262d", "labelBoxBorderColor": "#8b949e", "labelTextColor": "#e6edf3", "loopTextColor": "#e6edf3", "activationBkgColor": "#30363d", "activationBorderColor": "#8b949e"}}}%%
sequenceDiagram
    participant C as API Client
    participant GW as api-gateway (index.ts)
    participant UA as normalizeInboundUserAgent
    participant Auth as authenticateAndRateLimit
    participant Usage as recordUsage

    C->>GW: HTTP request (any protected path)
    GW->>GW: Resolve apiPath (infra routes exit early)
    GW->>UA: normalizeInboundUserAgent(User-Agent header)
    UA-->>GW: null (missing/blank/oversized) OR trimmed string

    alt "null OR length < 5"
        GW-->>C: 400 User-Agent must be 5-200 characters
    else valid UA
        GW->>GW: Check LEGACY_API_REDIRECT 308 if set
        GW->>Auth: authenticateAndRateLimit(..., inboundUserAgent)
        Auth->>Usage: recordUsage userAgent
        Auth-->>GW: AuthSuccess or 401/403/429
        GW-->>C: 200 / error response
    end
```

</a>
</details>

<sub>Reviews (2): Last reviewed commit: ["fix(api): address review feedback on Use..."](https://github.com/tarkovtracker-org/tarkovtracker/commit/6d85db4151bd6e13f4c5adc16b24ef86d684c9f4) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=45914259)</sub>

**Context used:**

- Context used - AGENTS.md ([source](https://app.greptile.com/dysektai/github/tarkovtracker-org/tarkovtracker/-/custom-context?memory=635c24fb-71e5-43a4-9979-74405b725e95))

### CODE REVIEW

File: workers/api-gateway/src/openapi.ts [hunk 1]
```diff
@@ -54,6 +57,19 @@ export const OPENAPI_SPEC = {
         description: 'Authorization: Bearer <token>',
       },
     },
+    parameters: {
+      UserAgentHeader: {
```
[the remark this answers is not in the document]
Comment (DysektAI) [reply workers/api-gateway/src/openapi.ts:62]: Fixed in 6d85db41. `UserAgentHeader` is now referenced via `$ref` in the `parameters` array of every protected operation (`/token`, `/progress`, `/team/progress`, `/progress/level/{level}`, `/progress/task/{taskId}`, `/progress/task/objective/{objectiveId}`, `/progress/tasks`). The shared `BadRequest` 400 response is also declared on the GET operations that previously omitted it. Infrastructure routes (`/health`, `/openapi.json`, `/docs`, `/robots.txt`) remain exempt. `pnpm --filter api-gateway run validate:openapi` passes.

File: workers/api-gateway/src/openapi.ts [hunk 2]
```diff
@@ -54,6 +57,19 @@ export const OPENAPI_SPEC = {
...
+        required: true,
+        description:
+          'A 5-200 character User-Agent identifying the client application, ' +
+          'e.g. "RatScanner/2.1 (+https://ratscanner.io)". ' +
+          'Requests outside that range are rejected with 400.',
+        schema: { type: 'string', minLength: 5, maxLength: 200 },
+        example: 'RatScanner/2.1 (+https://ratscanner.io)',
+      },
+    },
     responses: {
```
[the remark this answers is not in the document]
Comment (DysektAI) [reply workers/api-gateway/src/openapi.ts:74]: Fixed in 6d85db41. `UserAgentHeader` is now referenced via `$ref` in the `parameters` array of every protected operation (`/token`, `/progress`, `/team/progress`, `/progress/level/{level}`, `/progress/task/{taskId}`, `/progress/task/objective/{objectiveId}`, `/progress/tasks`). The shared `BadRequest` 400 response is also declared on the GET operations that previously omitted it. Infrastructure routes (`/health`, `/openapi.json`, `/docs`, `/robots.txt`) remain exempt. `pnpm --filter api-gateway run validate:openapi` passes.

File: workers/api-gateway/src/openapi.ts [hunk 3]
```diff
@@ -18,6 +18,9 @@ export const OPENAPI_SPEC = {
       'Token cap: each account may have at most 3 active API tokens. Revoke an existing ' +
       'token before creating a new one if the cap is reached. Token creation is only ' +
       'allowed through the token-create Edge Function and is rate-limited to 3/hour.\n\n' +
+      'User-Agent: a 5-200 character User-Agent header identifying the client application ' +
+      'is required on all API endpoints. Requests outside that range are rejected with 400. ' +
+      'Use a descriptive string like "AppName/1.0 (+https://your-app.com)".\n\n' +
```
[the remark this answers is not in the document]
Comment (DysektAI) [reply workers/api-gateway/src/openapi.ts]: Fixed in 6d85db41. `UserAgentHeader` is now referenced via `$ref` in the `parameters` array of every protected operation (`/token`, `/progress`, `/team/progress`, `/progress/level/{level}`, `/progress/task/{taskId}`, `/progress/task/objective/{objectiveId}`, `/progress/tasks`). The shared `BadRequest` 400 response is also declared on the GET operations that previously omitted it. Infrastructure routes (`/health`, `/openapi.json`, `/docs`, `/robots.txt`) remain exempt. `pnpm --filter api-gateway run validate:openapi` passes.

File: workers/api-gateway/src/openapi.ts [hunk 4]
```diff
@@ -54,6 +57,19 @@ export const OPENAPI_SPEC = {
         description: 'Authorization: Bearer <token>',
       },
     },
+    parameters: {
```
[the remark this answers is not in the document]
Comment (DysektAI) [reply workers/api-gateway/src/openapi.ts:61]: Fixed in 6d85db41. `UserAgentHeader` is now referenced via `$ref` in the `parameters` array of every protected operation (`/token`, `/progress`, `/team/progress`, `/progress/level/{level}`, `/progress/task/{taskId}`, `/progress/task/objective/{objectiveId}`, `/progress/tasks`). The shared `BadRequest` 400 response is also declared on the GET operations that previously omitted it. Infrastructure routes (`/health`, `/openapi.json`, `/docs`, `/robots.txt`) remain exempt. `pnpm --filter api-gateway run validate:openapi` passes.

File: workers/api-gateway/src/index.ts [hunk 5]
```diff
@@ -949,6 +954,20 @@ export default {
...
+      inboundUserAgent.length < INBOUND_USER_AGENT_MIN_LENGTH ||
+      (rawUserAgent?.trim().length ?? 0) > INBOUND_USER_AGENT_MAX_LENGTH
+    ) {
+      return errorResponse(
+        'User-Agent must be 5-200 characters (e.g. "AppName/1.0 (+https://your-app.com)")',
+        400,
+        origin,
+        reqOrigin
+      );
+    }
```
[the remark this answers is not in the document]
Comment (DysektAI) [reply workers/api-gateway/src/index.ts]: Fixed in 6d85db41. The User-Agent validation block now runs immediately after `apiPath` is resolved and before the legacy `LEGACY_API_REDIRECT` 308 branch, so `/api` and `/api/v2` requests validate missing/short/oversized User-Agent values before any redirect. A new gateway test (`validates User-Agent before issuing a legacy /api/v2 308 redirect`) confirms a too-short UA returns 400 instead of 308 when the redirect flag is on. The existing 308 redirect tests were updated to send a valid User-Agent.

File: workers/api-gateway/src/index.ts [hunk 6]
```diff
@@ -949,6 +954,20 @@ export default {
     if (!apiPath) {
       return new Response('Not Found', { status: 404, headers });
     }
+    const rawUserAgent = request.headers.get('User-Agent');
```
[the remark this answers is not in the document]
Comment (DysektAI) [reply workers/api-gateway/src/index.ts]: Fixed in 6d85db41. The User-Agent validation block now runs immediately after `apiPath` is resolved and before the legacy `LEGACY_API_REDIRECT` 308 branch, so `/api` and `/api/v2` requests validate missing/short/oversized User-Agent values before any redirect. A new gateway test (`validates User-Agent before issuing a legacy /api/v2 308 redirect`) confirms a too-short UA returns 400 instead of 308 when the redirect flag is on. The existing 308 redirect tests were updated to send a valid User-Agent.

File: workers/api-gateway/src/index.ts [hunk 7]
```diff
@@ -949,6 +954,20 @@ export default {
...
+    const inboundUserAgent = normalizeInboundUserAgent(rawUserAgent);
+    if (
+      !inboundUserAgent ||
+      inboundUserAgent.length < INBOUND_USER_AGENT_MIN_LENGTH ||
+      (rawUserAgent?.trim().length ?? 0) > INBOUND_USER_AGENT_MAX_LENGTH
+    ) {
```
[the remark this answers is not in the document]
Comment (DysektAI) [reply workers/api-gateway/src/index.ts]: Fixed in 6d85db41. `normalizeInboundUserAgent` now returns `null` for overlong values instead of truncating, so the caller's check collapses to `!inboundUserAgent || inboundUserAgent.length < MIN` with no coupling to the raw value. The signature also accepts `undefined` to satisfy strict callers. The unit test was updated to expect `null` for oversized input.

File: workers/api-gateway/src/index.ts [hunk 8]
```diff
@@ -949,6 +954,20 @@ export default {
     if (!apiPath) {
       return new Response('Not Found', { status: 404, headers });
     }
+    const rawUserAgent = request.headers.get('User-Agent');
+    const inboundUserAgent = normalizeInboundUserAgent(rawUserAgent);
+    if (
+      !inboundUserAgent ||
+      inboundUserAgent.length < INBOUND_USER_AGENT_MIN_LENGTH ||
+      (rawUserAgent?.trim().length ?? 0) > INBOUND_USER_AGENT_MAX_LENGTH
```
[the remark this answers is not in the document]
Comment (DysektAI) [reply workers/api-gateway/src/index.ts]: Fixed in 6d85db41. Added a gateway test (`rejects requests with an oversized User-Agent header`) that sends a 201-character User-Agent to `GET /token` and asserts a 400 with the expected error message, covering the gateway-level reject path rather than just the helper.

File: workers/api-gateway/src/openapi.ts [hunk 9]
```diff
@@ -18,6 +18,9 @@ export const OPENAPI_SPEC = {
       'Token cap: each account may have at most 3 active API tokens. Revoke an existing ' +
       'token before creating a new one if the cap is reached. Token creation is only ' +
       'allowed through the token-create Edge Function and is rate-limited to 3/hour.\n\n' +
+      'User-Agent: a 5-200 character User-Agent header identifying the client application ' +
+      'is required on all API endpoints. Requests outside that range are rejected with 400. ' +
```
[the remark this answers is not in the document]
Comment (DysektAI) [reply workers/api-gateway/src/openapi.ts]: Fixed in 6d85db41. The overview now reads "is required on protected API endpoints (token, progress, team). Infrastructure routes (/health, /openapi.json, /docs, /robots.txt) are exempt." so the published behavior matches the runtime exemption.

File: app/server/api/twitch/live.get.ts [hunk 10]
```diff
@@ -26,7 +27,11 @@ export default defineEventHandler(async (event) => {
   try {
     const response = await fetch(TWITCH_GQL_URL, {
       method: 'POST',
-      headers: { 'Client-ID': twitchClientId as string, 'Content-Type': 'application/json' },
+      headers: {
+        'Client-ID': twitchClientId as string,
+        'Content-Type': 'application/json',
+        'User-Agent': TARKOVTRACKER_USER_AGENT,
```
[the remark this answers is not in the document]
Comment (DysektAI) [reply app/server/api/twitch/live.get.ts:33]: Fixed in 6d85db41. The Twitch live test now imports `TARKOVTRACKER_USER_AGENT` and asserts `init.headers['User-Agent']` equals it, so a regression that drops the shared identifier is detected.

File: workers/api-gateway/src/__tests__/gateway.test.ts [hunk 11]
```diff
@@ -209,6 +214,29 @@ describe('api-gateway', () => {
...
   });
+  it('rejects requests with a missing User-Agent header', async () => {
+    const res = await worker.fetch(
+      new Request('https://api.tarkovtracker.org/token', {
+        method: 'GET',
+        headers: { Authorization: 'Bearer PVP_abc123' },
+      }),
+      BASE_ENV
+    );
+    const body = (await res.json()) as { success: boolean; error: string };
```
[the remark this answers is not in the document]
Comment (DysektAI) [reply workers/api-gateway/src/__tests__/gateway.test.ts:225]: Not a valid issue — the two `const body` declarations are in separate `it()` callback scopes (one in "rejects requests with a missing User-Agent header", one in "rejects requests with a too-short User-Agent header"), so there is no redeclaration. `pnpm run test:api-gateway` and `tsc --noEmit` both pass. Resolving as invalid.

**Comment by DysektAI** [comment]
## Review feedback addressed — commit `6d85db41`

All inline review threads from CodeRabbit, cubic, Greptile, and Kilo have been replied to with evidence. Summary of dispositions:

### Fixed
| Feedback | Source(s) | Fix |
|---|---|---|
| `UserAgentHeader` declared but never referenced in operations | kilo, greptile, coderabbit, cubic | Added `parameters: [{ $ref: '#/components/parameters/UserAgentHeader' }]` to every protected operation; added `400` response to the GET operations that omitted it. Infrastructure routes stay exempt. |
| User-Agent validation runs after the legacy 308 redirect, allowing bypass on `/api` and `/api/v2` | coderabbit, cubic | Moved the validation block to run immediately after `apiPath` is resolved and before the `LEGACY_API_REDIRECT` branch. Added a gateway test confirming a too-short UA returns 400 instead of 308 when the redirect flag is on. |
| `normalizeInboundUserAgent` clips overlong values, creating implicit coupling with the caller's max check | greptile | Helper now returns `null` for overlong input; the caller's check collapses to `!inboundUserAgent || length < MIN`. Signature also accepts `undefined`. |
| Oversized-header reject path has no gateway-level coverage | cubic | Added `rejects requests with an oversized User-Agent header` test (201-char UA → 400). |
| Overview says "all API endpoints" but health/docs bypass the gate | cubic | Narrowed to "protected API endpoints (token, progress, team). Infrastructure routes … are exempt." |
| Twitch request test doesn't assert the shared `User-Agent` | cubic | Test now imports `TARKOVTRACKER_USER_AGENT` and asserts the header on the upstream request. |
| Workers CI typecheck failure (`undefined` not assignable to `string \| null`) | CI | Fixed by widening `normalizeInboundUserAgent` to accept `undefined`. |

### Resolved as invalid
| Feedback | Source | Reason |
|---|---|---|
| Duplicate `const body` declarations in `gateway.test.ts` (lines 225 & 235) | coderabbit | The two declarations are in separate `it()` callback scopes — no redeclaration. `tsc --noEmit` and the test suite pass. |

### Validation run locally
- `pnpm --filter api-gateway run typecheck` — pass
- `pnpm --filter api-gateway run validate:openapi` — pass (schema valid)
- `pnpm run test:api-gateway` — 118/118 pass
- `pnpm run typecheck` (full) — pass
- `pnpm run lint` — pass (zero warnings, `app` scope)
- `prettier --check` on changed files — pass
- Husky pre-commit hook ran successfully (lint-staged + prettier + eslint)

Requesting another review cycle.

File: workers/api-gateway/src/openapi.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (DysektAI) [reply workers/api-gateway/src/openapi.ts:62]: Resolved.

File: workers/api-gateway/src/openapi.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (DysektAI) [reply workers/api-gateway/src/openapi.ts:74]: Resolved.

File: workers/api-gateway/src/index.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (DysektAI) [reply workers/api-gateway/src/index.ts]: Resolved.

File: workers/api-gateway/src/__tests__/gateway.test.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (DysektAI) [reply workers/api-gateway/src/__tests__/gateway.test.ts:225]: Resolved.

File: workers/api-gateway/src/index.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (DysektAI) [reply workers/api-gateway/src/index.ts]: Resolved.

File: workers/api-gateway/src/openapi.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (DysektAI) [reply workers/api-gateway/src/openapi.ts]: Resolved.

File: workers/api-gateway/src/openapi.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (DysektAI) [reply workers/api-gateway/src/openapi.ts:61]: Resolved.

File: workers/api-gateway/src/index.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (DysektAI) [reply workers/api-gateway/src/index.ts]: Resolved.

File: workers/api-gateway/src/openapi.ts (hunk 9 above)
[the remark this answers is not in the document]
Comment (DysektAI) [reply workers/api-gateway/src/openapi.ts]: Resolved.

File: app/server/api/twitch/live.get.ts (hunk 10 above)
[the remark this answers is not in the document]
Comment (DysektAI) [reply app/server/api/twitch/live.get.ts:33]: Resolved.

File: workers/api-gateway/src/index.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (DysektAI) [reply workers/api-gateway/src/index.ts]: Resolved.

File: workers/api-gateway/src/__tests__/gateway.test.ts [hunk 12]
```diff
@@ -209,12 +214,62 @@ describe('api-gateway', () => {
...
+  it('rejects requests with an oversized User-Agent header', async () => {
+    const res = await worker.fetch(
+      buildRequest('/token', {
+        method: 'GET',
+        headers: { Authorization: 'Bearer PVP_abc123', 'User-Agent': 'x'.repeat(201) },
+      }),
+      BASE_ENV
+    );
+    expect(res.status).toBe(400);
+    const body = (await res.json()) as { success: boolean; error: string };
```
[the remark this answers is not in the document]
Comment (DysektAI) [reply workers/api-gateway/src/__tests__/gateway.test.ts:249]: Not a real issue — each `const body` lives in its own `it()`/`test()` block scope (lines 178, 186, 193, 225, 235, 249, 264, 322, 695, 932 in `workers/api-gateway/src/__tests__/gateway.test.ts`), so there is no redeclaration in any shared scope. CI `Type Check` on this PR is green (conclusion: SUCCESS), which would fail on a real TS2440/2451 duplicate-binding error. The `Also applies to: 264-264` note points at the same pattern inside a separate `it()` block. Resolving as invalid.
